stingray/covariancespectrum.py

```diff
@@ -189,6 +206,23 @@ def _construct_energy_covar(self):
 
             self.energy_covar[energy] = covar
 
+            # Calculate error
+            if self.error is None:
+                std = np.mean(lc_ref)**0.5
+            elif type(self.error) in [list, np.array]: # Array of numbers
+                std = np.mean(self.error)
```

Comment:
In general, I would use `isinstance()` to check for types instead of `type`, because the latter doesn't always work as expected.

I think we might want to test here whether this is generally an iterable, not a specific type of `list` or `np.array`. 
I've just been told that the way to test generally for iterables could be this:
`if hasattr(x, '__iter__'):`
`    ...`

Basically, all iterables need to implement `__iter__`, so testing whether this exists is a good way to do this kind of test.